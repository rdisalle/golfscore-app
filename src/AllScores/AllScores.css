.AllScores_list {
    color: black;
    list-style-type: none;
    padding: 10px;
    margin-right: 10px;
    margin-left: 10px;
    margin-top: 10px;
}

.AllScores_score {
    border: 1px solid #4681A6;
    margin-top: 25px;
    margin-bottom: 25px;
    padding: 10px;
}

.AllScores_course {
    color: #025918;
    font-size: 25px;
}

.AllScores_add {
    color: #4681A6;
    font-size: 20px;
    text-decoration: none;
}

.AllScores_viewLink {
    color: #025918;
    text-decoration: none;
    border: 1px #4681A6 solid;
    padding: 5px;
    padding-right: 95px;
    padding-left: 95px;
    font-size: 15px;
}

@media only screen and (min-device-width: 768px) {
    .AllScores_list {
        display: flex;
        flex-direction: row;
        justify-content: space-evenly;
        flex-wrap: wrap;
    }
    .AllScores_score {
        width: 300px;
    }
}

@media screen and (min-device-width: 1200px) {
    .AllScores_list {
        display: flex;
        flex-direction: row;
        justify-content: space-evenly;
        flex-wrap: wrap;
    }
    .AllScores_score {
        width: 500px;
    }
}