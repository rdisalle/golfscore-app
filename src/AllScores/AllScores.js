import React, { Component } from 'react';
import './AllScores.css';
import ScoresPage from '../ScoresPage/ScoresPage'
import { Link } from 'react-router-dom';
import GolfContext from '../GolfContext'

class AllScores extends Component {
    static defaultProps = {
        match: {
          params: {}
        }
      }

    static contextType = GolfContext;

    onDelete = () => {
        this.props.history.push('/all-scores/:courseId')
      }
    
    getScoresForCourse = (scores=[], courseId) => (
        (!courseId)
            ? scores
            : scores.filter(score => score.course_id === JSON.parse(courseId))
)

    render() {
        const { courseId } = this.props.match.params
        const { scores=[] } = this.context
        const scoresForCourse = this.getScoresForCourse(scores, courseId)
        console.log(`${courseId}`)
        return (
            <div className="AllScores">
                <ul className="AllScores_list">
                    {scoresForCourse.map(score =>
                    <li key={score.id}>
                        <h2 className="CourseName">
                            {score.course}
                        </h2>
                        <ScoresPage
                            id={score.id}
                            name={score.name}
                            total_score={score.total_score}
                            to_par={score.to_par}
                        />
                        <Link to={`/course-score/${score.id}`}>
                            View 
                        </Link>
                    </li>
                    )}
                </ul>
                <Link to={`/add-score-page/${courseId}`}>
                    Add Score
                </Link>
            </div>
        );
    }
}

AllScores.propTypes = {
    match: PropTypes.object
  };

export default AllScores;