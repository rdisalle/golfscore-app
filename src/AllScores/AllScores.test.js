import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import AllScores from './Allscores'

describe(`AllScores component`, () => {
  const props = {
    scores: [
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Golf Course Score One",
        "course": "Golf Course",
        "course_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "score_hole_one": "3", 
        "score_hole_two": "4", 
        "score_hole_three": "5",
        "score_hole_four": "4",
        "score_hole_five": "5", 
        "score_hole_six": "6", 
        "score_hole_seven": "2",
        "score_hole_eight": "4",
        "score_hole_nine": "4",
        "score_hole_ten": "5",
        "score_hole_eleven": "5", 
        "score_hole_twelve": "5",
        "score_hole_thirteen": "4",
        "score_hole_fourteen": "4", 
        "score_hole_fifteen": "4", 
        "score_hole_sixteen": "4",
        "score_hole_seventeen": "4", 
        "score_hole_eighteen": "4",
        "total_score": "80",
        "to_par": "+8"
      },
      {
        "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Golf Course Score Two",
        "course": "Golf Course",
        "course_id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "score_hole_one": "3", 
        "score_hole_two": "4", 
        "score_hole_three": "5",
        "score_hole_four": "4",
        "score_hole_five": "5", 
        "score_hole_six": "6", 
        "score_hole_seven": "2",
        "score_hole_eight": "4",
        "score_hole_nine": "4",
        "score_hole_ten": "5",
        "score_hole_eleven": "5", 
        "score_hole_twelve": "5",
        "score_hole_thirteen": "4",
        "score_hole_fourteen": "4", 
        "score_hole_fifteen": "4", 
        "score_hole_sixteen": "4",
        "score_hole_seventeen": "4", 
        "score_hole_eighteen": "4",
        "total_score": "80",
        "to_par": "+8"
      },
      {
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Golf Course Score One",
        "course": "Golf Course",
        "course_id": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "score_hole_one": "3", 
        "score_hole_two": "4", 
        "score_hole_three": "5",
        "score_hole_four": "4",
        "score_hole_five": "5", 
        "score_hole_six": "6", 
        "score_hole_seven": "2",
        "score_hole_eight": "4",
        "score_hole_nine": "4",
        "score_hole_ten": "5",
        "score_hole_eleven": "5", 
        "score_hole_twelve": "5",
        "score_hole_thirteen": "4",
        "score_hole_fourteen": "4", 
        "score_hole_fifteen": "4", 
        "score_hole_sixteen": "4",
        "score_hole_seventeen": "4", 
        "score_hole_eighteen": "4",
        "total_score": "80",
        "to_par": "+8"
      },
      {
        "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Golf Course Score One",
        "course": "Golf Course",
        "course_id": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "score_hole_one": "3", 
        "score_hole_two": "4", 
        "score_hole_three": "5",
        "score_hole_four": "4",
        "score_hole_five": "5", 
        "score_hole_six": "6", 
        "score_hole_seven": "2",
        "score_hole_eight": "4",
        "score_hole_nine": "4",
        "score_hole_ten": "5",
        "score_hole_eleven": "5", 
        "score_hole_twelve": "5",
        "score_hole_thirteen": "4",
        "score_hole_fourteen": "4", 
        "score_hole_fifteen": "4", 
        "score_hole_sixteen": "4",
        "score_hole_seventeen": "4", 
        "score_hole_eighteen": "4",
        "total_score": "80",
        "to_par": "+8"
      },
    ]
  }

  it('renders a .AllScores by default', () => {
    const wrapper = shallow(<AllScores />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Score in ul for each score in array', () => {
    const ul = shallow(<AllScores {...props} />)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
