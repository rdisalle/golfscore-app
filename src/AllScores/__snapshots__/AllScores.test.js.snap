// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllScores component renders a .AllScores by default 1`] = `
<div
  className="AllScores"
>
  <ul
    className="AllScores_list"
  />
  <Link
    to="/add-score-page/undefined"
  >
    Add Score
  </Link>
</div>
`;

exports[`AllScores component renders a Score in ul for each score in array 1`] = `
<ul
  className="AllScores_list"
/>
`;
